Makefile

```diff
@@ -21,31 +21,114 @@ for file in /etc/os-release /usr/lib/os-release; do \
     fi \
 done)
 
-# A "CC system build" is either triggered by specifying
-# a special target or via an environment variable.
+#------------------------------
+# project-specifics
+
+# build type for Clear Containers
+CC_TYPE = cc
+
+CC_PROJECT_NAME = Intel® Clear Containers
+CC_PROJECT_TAG = clear-containers
+CC_PROJECT_URL = https://github.com/clearcontainers
+
+# build type for Kata Containers
+KATA_TYPE = kata
+
+KATA_PROJECT_NAME = Kata Containers
+KATA_PROJECT_TAG = kata-containers
+KATA_PROJECT_URL = https://github.com/kata-containers
+
+#------------------------------
+
+# If this environment variable is set to any value,
+# enable the Clear Containers system build.
+ifneq (,$(CC_SYSTEM_BUILD))
+    system_build_type = $(CC_TYPE)
+endif
+
+# If this environment variable is set to any value,
+# enable the Kata Containers system build.
+ifneq (,$(KATA_SYSTEM_BUILD))
+    system_build_type = $(KATA_TYPE)
+endif
+
+ifneq ($(SYSTEM_BUILD_TYPE),)
+    ifeq ($(SYSTEM_BUILD_TYPE),$(CC_TYPE))
+        system_build_type = $(SYSTEM_BUILD_TYPE)
```

Comment:
Although equivalent, I think it looks less confusing to do:
```
system_build_type = $(CC_TYPE)
```
here, and the equiv for KATA below.

I guess all we are really doing here is checking that the `SYSTEM_BUILD_TYPE` is actually a type we know about. In which case, would an error case catchall of something like `ifeq ($(SYSTEM_BUILD_TYPE),)` after these two cases make sense? (that is, nobody set the type?) But, maybe that does not mesh with the other methods of choosing which sort of build (I sort of wish we only had one method....)